tensorflow_recommenders_addons/dynamic_embedding/python/ops/math_ops.py

```diff
@@ -127,12 +152,189 @@ def _sparse_segment_sum_gpu(data,
+        tf_logging.warn('`tfra.dynamic_embedding.sparse_fill_empty_rows` is not'
+                        ' found. Use tf.sparse.fill_empty_rows instead.')
+        return tf.sparse.fill_empty_rows(sp_input, default_value, name=name)
+
+    else:
+      predef = _sparse_fill_empty_rows_gpu(sp_input, default_value, name=name)
+      use_origin = True
+      if predef[0].values.device == '':
+        tf_logging.warn(
+            'Haven\'t specify devices while GPU devices are'
+            'available: {}, use CPU by default.'.format(gpu_devices))
+      else:
+        device_type = predef[0].values.device.split(':')[-2][-3:].lower()
+        if device_type == 'gpu':
+          use_origin = False
+
+      if use_origin:
+        return tf.sparse.fill_empty_rows(sp_input, default_value, name=name)
+      return predef
+
+  else:
+    return tf.sparse.fill_empty_rows(sp_input, default_value, name=name)
+
+
+def _sparse_fill_empty_rows_gpu(sp_input, default_value, name=None):
+  if not hasattr(tfra_math_ops, 'tfra_sparse_fill_empty_rows'):
+    tf_logging.warn('`tfra.dynamic_embedding.sparse_fill_empty_rows` is not'
+                    ' found. Use tf.sparse.fill_empty_rows instead.')
+    return tf.sparse.fill_empty_rows(sp_input, default_value, name=name)
+
+  sp_input = _convert_to_sparse_tensor(sp_input)
+  with ops.name_scope(name, "SparseFillEmptyRows", [sp_input]):
+    default_value = ops.convert_to_tensor(default_value,
+                                          dtype=sp_input.values.dtype)
+    (output_indices, output_values, empty_row_indicator,
+     unused_reverse_index_map) = tfra_math_ops.tfra_sparse_fill_empty_rows(
+         indices=sp_input.indices,
+         values=sp_input.values,
+         dense_shape=sp_input.dense_shape,
+         default_value=default_value)
+    return (sparse_tensor.SparseTensor(indices=output_indices,
+                                       values=output_values,
+                                       dense_shape=sp_input.dense_shape),
+            empty_row_indicator)
+
+
+def sparse_reshape(sp_input, shape, name=None):
+  """Reshapes a `SparseTensor` to represent values in a new dense shape.
+
+  It do same things as `tf.sparse.reshape`. Here we provide GPU impl.
+
+  Go [tf api](https://www.tensorflow.org/api_docs/python/tf/sparse/reshape)
+  for more details.
+
+  Args:
+    sp_input: The input `SparseTensor`.
+    shape: A 1-D (vector) int64 `Tensor` specifying the new dense shape of the
+      represented `SparseTensor`.
+    name: A name prefix for the returned tensors (optional)
+
+  Returns:
+    A `SparseTensor` with the same non-empty values but with indices calculated
+    by the new dense shape.
+  """
+  gpu_devices = config.list_physical_devices('GPU')
+  if gpu_devices:
+    if context.executing_eagerly():
+      try:
+        return _sparse_reshape_gpu(sp_input, shape, name=name)
+      except errors.NotFoundError:
+        tf_logging.warn('`tfra.dynamic_embedding.sparse_reshape` is not'
```

Comment:
Maybe this log will be printed too frequently on eager mode ? @Lifann

Reply:
It only happens when trying to run the TFRA CPU version on GPU enabled environment. It reminds someone to install the GPU version when it is enabled.